src/kernel/kmain.zig

```diff
@@ -106,6 +107,14 @@ export fn kmain(boot_payload: arch.BootPayload) void {
 
     if (rd_module) |module| {
         // Load the ram disk
+        // Create a RamDevice
+        // const rd_len: usize = module.region.end - module.region.start;
+        // const ramdisk_bytes = @intToPtr([*]u8, module.region.start)[0..rd_len];
+        // Tbh, there is a lot, A LOT of memory coping. It would be faster to just use the ram disk in memory already.
+        // It would be nice to use the ram device for the initrd but would but unnecessary.
+        // var ram_disk_device = ram_device.RamDevice.init(ramdisk_bytes, &kernel_heap.allocator) catch |e| {
+        //     panic_root.panic(@errorReturnTrace(), "Failed to initialise ramdisk device: {}\n", .{e});
+        // };
```

Comment:
Here there be commented code.

Reply:
I was going to use the new RamDevice, but there just seems to be too much allocations for a RAM device. Should the read function just return a slice of the memory instead of allocating it?

Reply:
The initrd read function or the RamDevice read function?

Reply:
The RamDevice read function, as it allocated memory for the read.